### PR TITLE
52 fix endpoint schemathesis tests

### PR DESCRIPTION
**Pull request by thulasiramanmt**
# Pull request

**Resolves:** #52

### CODE REVIEW

File: darkmoon/api/backend/endpoint.py [hunk 1]
```diff
@@ -1,68 +1,151 @@
...
 
 router = APIRouter(prefix="/endpoints", tags=["endpoints"])
 
 
-@router.get("/metadata")
+async def called_process_error_handler(
+    request: Request,
+    exc: subprocess.CalledProcessError,
+) -> JSONResponse:
+    """Docstring."""
```
Comment (pelavarthi) [inline darkmoon/api/backend/endpoint.py:29]: Missing docstring

File: darkmoon/api/backend/endpoint.py [hunk 2]
```diff
@@ -1,68 +1,151 @@
...
+    "/metadata",
+    responses={
+        400: {"Client Error Response": "Bad Request"},
+    },
+)
 async def get_metadata_endpoint(
-    file: PyPath = Query(..., description="Path to the file"),
-    source_iso: PyPath = Query(..., description="Path to the source ISO"),
+    file: UploadFile = File(...),
+    source_iso: UploadFile = File(...),
```
Comment (pelavarthi) [inline darkmoon/api/backend/endpoint.py:44]: Check object type for source_iso. Might need to make it a string.

File: darkmoon/common/utils.py [hunk 3]
```diff
@@ -116,7 +118,9 @@ def extract_files(file: Path, source_iso: Path, url: str) -> None:
     """Extract vmdk and put in new folder."""
     with tempfile.TemporaryDirectory() as tmpdirname:
         cmd = ["7z", "x", str(file), "-o" + tmpdirname]
-        subprocess.run(cmd, check=True)
+        result = subprocess.run(cmd)
+        if result.returncode != 0:
```
Comment (pelavarthi) [inline darkmoon/common/utils.py:122]: Should it be result.status code?

**Review by pelavarthi — APPROVED** [review]

**Review by pelavarthi — DISMISSED** [review]

File: darkmoon/api/backend/endpoint.py [hunk 4]
```diff
@@ -1,68 +1,151 @@
...
+        400: {"Client Error Response": "Bad Request"},
+        422: {"Client Error Response": "Unprocessable Content"},
+        500: {"Server Error Response": "Internal Server Error"},
+        504: {"Server Error Response": "Gateway Timeout"},
+    },
+)
 async def extract_files_endpoint(
-    file: UploadFile = File(...),
+    file: UploadFile = File(...),  # only vmdks
     source_iso: PyPath = PyPath("..."),
```
Comment (DoodlePooge) [inline darkmoon/api/backend/endpoint.py:112]: How come some parameters for source_iso take in a path while other take UploadFile?

File: darkmoon/api/backend/endpoint.py [hunk 5]
```diff
@@ -1,68 +1,151 @@
...
+@router.post(
+    "/extract-files",
+    responses={
+        400: {"Client Error Response": "Bad Request"},
+        422: {"Client Error Response": "Unprocessable Content"},
+        500: {"Server Error Response": "Internal Server Error"},
+        504: {"Server Error Response": "Gateway Timeout"},
+    },
+)
 async def extract_files_endpoint(
```
Comment (DoodlePooge) [inline darkmoon/api/backend/endpoint.py:110]: This endpoint fails with the proper parameters provided.

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

**Review by DoodlePooge — DISMISSED** [review]

**Review by DoodlePooge — APPROVED** [review]

**Pull request merged**